src/services/feature/feature.service.ts

```
@@ -0,0 +1,68 @@
+      console.error(
+        "Only one addon of the same feature is allowed to be added, please remove the duplicate addon"
+      );
+      throw new Error(
+        "Only one addon of the same feature is allowed to be added, please remove the duplicate addon"
+      );
+    }
+
+    const planFeatures: Record<string, Feature> = plan.features as any;
+    const addonFeatures: Record<string, Feature> = addons?.reduce(
+      (acc, { key, feature }) => ({
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)